applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_sink.c

```diff
@@ -81,6 +80,89 @@ static le_audio_receive_cb receive_cb;
 static bool init_routine_completed;
 static bool paused;
 
+static struct bt_csip_set_member_svc_inst *csip;
+
+static uint8_t flags_adv_data;
+static uint8_t gap_appear_adv_data[BT_UUID_SIZE_16];
+static uint8_t csip_rsi_adv_data[BT_CSIP_RSI_SIZE];
+
+#define CSIP_SET_SIZE 2
+enum csip_set_rank {
+	CSIP_HL_RANK = 1,
+	CSIP_HR_RANK = 2
+};
+
+/* Callback for locking state change from server side */
+static void csip_lock_changed_cb(struct bt_conn *conn, struct bt_csip_set_member_svc_inst *csip,
+				 bool locked)
+{
+	LOG_DBG("Client %p %s the lock", (void *)conn, locked ? "locked" : "released");
+}
+
+/* Callback for SIRK read request from peer side */
+static uint8_t sirk_read_req_cb(struct bt_conn *conn, struct bt_csip_set_member_svc_inst *csip)
+{
+	/* Accept the request to read the SIRK, but return encrypted SIRK instead of plaintext */
+	return BT_CSIP_READ_SIRK_REQ_RSP_ACCEPT_ENC;
+}
+
+static struct bt_csip_set_member_cb csip_callbacks = {
+	.lock_changed = csip_lock_changed_cb,
+	.sirk_read_req = sirk_read_req_cb,
+};
+
+struct bt_csip_set_member_register_param csip_param = {
+	.set_size = CSIP_SET_SIZE,
+	.lockable = true,
+	.cb = &csip_callbacks,
+};
+
+static uint8_t broadcast_sink_adv_data[] = {
+	BT_UUID_16_ENCODE(BT_UUID_BASS_VAL),
+	BT_UUID_16_ENCODE(BT_UUID_PACS_VAL),
+};
```

Comment:
Shouldn't this be part of the uuid_buf? I think there should be a function similar to unicast_server_uuid_populate (this is not called btw, I think that is a bug in the unicast_server)